src/VirtoCommerce.Platform.Web/wwwroot/js/app/exportImport/blades/import-main.js

```diff
@@ -1,125 +1,129 @@
 angular.module('platformWebApp')
-.controller('platformWebApp.exportImport.importMainController', ['$scope', 'platformWebApp.bladeNavigationService', 'platformWebApp.exportImport.resource', 'FileUploader', function ($scope, bladeNavigationService, exportImportResourse, FileUploader) {
-    var blade = $scope.blade;
-    blade.updatePermission = 'platform:exportImport:import';
-    blade.headIcon = 'fa fa-download';
-    blade.title = 'platform.blades.import-main.title';
-    blade.isLoading = false;
-    $scope.importRequest = {};
-    var origManifest;
-
-    $scope.$on("new-notification-event", function (event, notification) {
-        if (blade.notification && notification.id == blade.notification.id) {
-            angular.copy(notification, blade.notification);
-            if (notification.errorCount > 0) {
-                bladeNavigationService.setError('Import error', blade);
+    .controller('platformWebApp.exportImport.importMainController', ['$scope', 'platformWebApp.bladeNavigationService', 'platformWebApp.exportImport.resource', 'FileUploader', function ($scope, bladeNavigationService, exportImportResourse, FileUploader) {
+        var blade = $scope.blade;
+        blade.updatePermission = 'platform:exportImport:import';
+        blade.headIcon = 'fa fa-download';
+        blade.title = 'platform.blades.import-main.title';
+        blade.isLoading = false;
+        $scope.importRequest = {};
+        var origManifest;
+
+        $scope.$on("new-notification-event", function (event, notification) {
+            if (notification.jobId && notification.finished) {
+                blade.toolbarCommands.splice(0, 1, commandClose);
```

Comment:
1. The useless operation, look on Line 33, did you suppose to `return`?
2. Extract method.

Reply:
Git was layer this.